apps/frontend/src/pages/access-management/access-management.tsx

```diff
@@ -1,54 +1,73 @@
-import {
-	PageLayout,
-	Table,
-	TablePagination,
-} from "~/libs/components/components.js";
+import { Button, Modal, PageLayout } from "~/libs/components/components.js";
 import { DataStatus } from "~/libs/enums/enums.js";
 import {
 	useAppDispatch,
 	useAppSelector,
+	useCallback,
 	useEffect,
+	useModal,
 	usePagination,
 } from "~/libs/hooks/hooks.js";
-import { actions as groupActions } from "~/modules/groups/groups.js";
+import {
+	actions as groupActions,
+	type GroupCreateRequestDto,
+} from "~/modules/groups/groups.js";
 import { actions as userActions } from "~/modules/users/users.js";
 
-import {
-	getGroupColumns,
-	getGroupRows,
-	getUserColumns,
-	getUserRows,
-} from "./libs/helpers/helpers.js";
-import { type GroupRow, type UserRow } from "./libs/types/types.js";
+import { GroupCreateForm } from "./components/group-create-form/group-create-form.js";
+import { GroupsTable } from "./components/groups-table/groups-table.js";
+import { UsersTable } from "./components/users-table/users-table.js";
 import styles from "./styles.module.css";
 
 const AccessManagement = (): JSX.Element => {
 	const dispatch = useAppDispatch();
 
 	const {
+		data: users,
 		dataStatus: usersDataStatus,
-		users,
-		usersTotalCount,
-	} = useAppSelector(({ users }) => users);
+		totalCount: usersTotalCount,
+	} = useAppSelector(({ users }) => users.users);
 
 	const { dataStatus: groupsDataStatus, groups } = useAppSelector(
 		({ groups }) => groups,
 	);
 
-	const { onPageChange, onPageSizeChange, page, pageSize } = usePagination({
+	const {
+		onPageChange: onUserPageChange,
+		onPageSizeChange: onUserPageSizeChange,
+		page: userPage,
+		pageSize: userPageSize,
+	} = usePagination({
 		totalItemsCount: usersTotalCount,
 	});
 
+	const {
+		isOpened: isModalOpened,
+		onClose: onModalClose,
+		onOpen: onModalOpen,
+	} = useModal();
+
 	useEffect(() => {
-		void dispatch(userActions.loadAll({ page, pageSize }));
-		void dispatch(groupActions.loadAll());
-	}, [dispatch, page, pageSize]);
+		void dispatch(
+			userActions.loadAll({ page: userPage, pageSize: userPageSize }),
+		);
 
-	const userColumns = getUserColumns();
-	const userData: UserRow[] = getUserRows(users);
+		void dispatch(groupActions.loadAll());
+	}, [dispatch, userPage, userPageSize]);
 
-	const groupColumns = getGroupColumns();
-	const groupData: GroupRow[] = getGroupRows(groups);
+	const handleGroupCreateSubmit = useCallback(
+		(payload: GroupCreateRequestDto): void => {
+			dispatch(groupActions.create(payload))
+				.then(() => {
+					onModalClose();
+					void dispatch(
+						userActions.loadAll({ page: userPage, pageSize: userPageSize }),
+					);
+				})
+				.catch(() => {});
```

Comment:
We shouldn't await dispatch. `userActions.loadAll` can be dispatched inside `groupActions.create`. For modal flow, take a look at how it's done with status during project creation
```suggestion
```

Reply:
So this is what's needed?

` const create = createAsyncThunk<
	GroupCreateResponseDto,
	{ payload: GroupCreateRequestDto; query: PaginationQueryParameters },
	AsyncThunkConfig
>(${sliceName}/create, async ({ payload, query }, { dispatch, extra }) => {
	const { groupApi, toastNotifier } = extra;

	const response = await groupApi.create(payload);

	toastNotifier.showSuccess(NotificationMessage.GROUP_CREATE_SUCCESS);

	void dispatch(userActions.loadAll(query));

	return response;
}); `

Reply:
yes